tests/e2e_test.go

```diff
@@ -31,38 +33,86 @@ var _ = Describe("ske-operator helm chart", func() {
+
+				By("deploying kratix")
+				//if the Kratix got created successfully by helm install, this means the
+				//webhook was running successfully
+				run("kubectl", context, "get", "kratixes", "kratix")
+				kratixVersion = run("kubectl", context, "get", "kratixes", "kratix", "-o", "jsonpath={.spec.version}")
+				creationTimestamp = run("kubectl", context, "get", "kratixes", "kratix", "-o", "jsonpath={.metadata.creationTimestamp}")
+
+				By("creating any additional resources provided in values file")
+				run("kubectl", context, "get", "secrets", "git-credentials", "-n=default")
+				run("kubectl", context, "get", "gitstatestores", "default")
+				run("kubectl", context, "get", "destinations", "worker-1")
+			})
+
+			By("upgrading SKE", func() {
+				By("upgrading to update additional resources", func() {
+					run("pwd")
+					run("helm", "upgrade", "ske-operator", "../ske-operator/", "-n=kratix-platform-system",
+						"-f=./assets/values-with-updated-additional-resources.yaml", "--set-string", "skeLicense="+skeLicenseToken, "--wait")
+
+					encodedUsername := run("kubectl", context, "get", "secrets", "git-credentials", "-o", "jsonpath={.data.username}")
+					decodedUsername, err := base64.StdEncoding.DecodeString(encodedUsername)
+					Expect(err).NotTo(HaveOccurred())
+					Expect(string(decodedUsername)).To(Equal("now"))
+				})
+
+				By("upgrading to delete additional resources", func() {
+					run("pwd")
+					run("helm", "upgrade", "ske-operator", "../ske-operator/", "-n=kratix-platform-system",
+						"-f=./assets/values-with-deleted-additional-resources.yaml", "--set-string", "skeLicense="+skeLicenseToken, "--wait")
+
+					Expect(run("kubectl", context, "get", "secrets", "git-credentials", "--ignore-not-found")).To(BeEmpty())
+				})
+
+				By("upgrading to update SKE additional resources", func() {
+					run("pwd")
+					run("helm", "upgrade", "ske-operator", "../ske-operator/", "-n=kratix-platform-system",
+						"-f=./assets/values-with-updated-ske-additional-resources.yaml", "--set-string", "skeLicense="+skeLicenseToken, "--wait")
+
+					Expect(run("kubectl", context, "get", "gitstatestores.platform.kratix.io", "default", "-o", "jsonpath={.spec.branch}")).To(Equal("prod-branch"))
+					Expect(run("kubectl", context, "get", "destinations.platform.kratix.io", "worker-1", "-o", "jsonpath={.metadata.labels.environment}")).To(Equal("prod"))
+				})
+
+				By("upgrading SKE version", func() {
+					run("pwd")
+					run("helm", "upgrade", "ske-operator", "../ske-operator/", "-n=kratix-platform-system",
+						"-f=./assets/values-with-upgrade.yaml", "--set-string", "skeLicense="+skeLicenseToken, "--wait")
+
+					Expect(run("kubectl", context, "get", "kratixes", "kratix", "-o", "jsonpath={.metadata.creationTimestamp}")).To(Equal(creationTimestamp))
+					Expect(run("kubectl", context, "get", "kratixes", "kratix", "-o", "jsonpath={.spec.version}")).NotTo(Equal(kratixVersion))
+				})
```

Comment:
could all of this be achieved via a single upgrade?

i.e.: a single upgrade could:
* bump ske version
* createa a new / update or delet ean existing resource

we should also probably assert on the desired state of the entire "additional resources" block at this as well.

Reply:
Yes, makes sense. 👍 

> we should also probably assert on the desired state of the entire "additional resources" block at this as well.

Could you elaborate this? ☝️